File: herencia-dto/AMarotoChivite-HerenciaDTO/AMaroto-Chivite-HerenciasDto-Java/src/main/java/Ficheros/HerenciasDto/models/dto/PersonListDTO.java
package Ficheros.HerenciasDto.models.dto;

import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;

import java.util.List;

@Root(name = "listPersonas")
public class PersonListDTO {

    private List<PersonDTO> myList;

    @ElementList(name = "lista_alumnos", entry = "alumno", inline = true)
    private List<PersonDTO.Alumno> myListAlumno;

    @ElementList(name = "lista_profesores", entry = "profesor", inline = true)
    private List<PersonDTO.Profesor> myListProfesor;

    public PersonListDTO() {
    }

    public PersonListDTO(List<PersonDTO.Alumno> myListAlumno, List<PersonDTO.Profesor> myListProfesor) {
        this.myListAlumno = myListAlumno;
        this.myListProfesor = myListProfesor;
    }

    public List<PersonDTO> getMyList() {
        return myList;
    }

    public void setMyList(List<PersonDTO> myList) {
        this.myList = myList;
    }

    public List<PersonDTO.Alumno> getMyListAlumno() {
        return myListAlumno;
    }

    public void setMyListAlumno(List<PersonDTO.Alumno> myListAlumno) {
        this.myListAlumno = myListAlumno;
    }

    public List<PersonDTO.Profesor> getMyListProfesor() {
        return myListProfesor;
    }

    public void setMyListProfesor(List<PersonDTO.Profesor> myListProfesor) {
        this.myListProfesor = myListProfesor;
    }

}

File: herencia-dto/AMarotoChivite-HerenciaDTO/AMarotoChivite-HerenciasDto-Kotlin/src/main/kotlin/Ficheros/HerenciasDto/models/dto/PersonListDTO.kt
package Ficheros.HerenciasDto.models.dto

import org.simpleframework.xml.ElementList
import org.simpleframework.xml.Root

@Root(name = "listPersonas") // Para XML
class PersonListDTO(
    var myListAlumno: List<PersonDTO.Alumno>? = null,
    var myListProfesor: List<PersonDTO.Profesor>? = null
) {
    constructor(
        myListAlumno: List<PersonDTO.Alumno>?, myListProfesor: List<PersonDTO.Profesor>?, myList: List<PersonDTO>
    ) : this(myListAlumno, myListProfesor) {
        this.myList = myList
    }

    @field:ElementList(name = "lista_personas", entry = "persona", inline = true) // Para XML
    var myList: List<PersonDTO>? = null // En el caso de XML no habría ningún problema podemos utilizar el general
}